homeassistant/components/eq3btsmart/config_flow.py

```diff
@@ -0,0 +1,100 @@
+"""Config flow for eQ-3 Bluetooth Smart thermostats."""
+
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import BluetoothServiceInfoBleak
+from homeassistant.const import CONF_MAC
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.util import slugify
+
+from .const import DOMAIN
+from .schemas import SCHEMA_MAC
+
+
+class EQ3ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for eQ-3 Bluetooth Smart thermostats."""
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+
+        self.mac_address: str = ""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        errors: dict[str, str] = {}
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=SCHEMA_MAC,
+                errors=errors,
+            )
+
+        mac_address = format_mac(user_input[CONF_MAC])
+
+        if not validate_mac(mac_address):
+            errors[CONF_MAC] = "invalid_mac_address"
+            return self.async_show_form(
+                step_id="user",
+                data_schema=SCHEMA_MAC,
+                errors=errors,
+            )
+
+        await self.async_set_unique_id(mac_address)
+        self._abort_if_unique_id_configured(updates=user_input)
+
+        # We can not validate if this mac actually is an eQ-3 thermostat,
+        # since the thermostat probably is not advertising right now.
+        return self.async_create_entry(title=slugify(mac_address), data=user_input)
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> FlowResult:
+        """Handle bluetooth discovery."""
+
+        self.mac_address = format_mac(discovery_info.address)
+
+        await self.async_set_unique_id(self.mac_address)
+        self._abort_if_unique_id_configured()
+
+        self.context.update({"title_placeholders": {CONF_MAC: self.mac_address}})
+
+        return await self.async_step_init()
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle flow start."""
+
+        self._async_abort_entries_match({CONF_MAC: self.mac_address})
```

Comment:
It looks like we don't need to store `CONF_MAC` since its the same as the unique id